phoenixRest/models/core/agenda_entry.py

```diff
@@ -30,22 +30,80 @@ class AgendaEntry(Base):
     event_uuid = Column(UUID(as_uuid=True), ForeignKey("event.uuid"), nullable=False)
     event = relationship("Event")
     
-    time = Column(DateTime, nullable=False)
-
     title = Column(Text, nullable=False)
     description = Column(Text, nullable=False)
+    location = Column(Text, nullable=True)
+    time = Column(DateTime, nullable=False)
+
+    deviating_time = Column(DateTime, nullable=True)
+    deviating_location = Column(Text, nullable=True)
+    deviating_information = Column(Text, nullable=True)
+
+    state_pinned = Column(Boolean, server_default='false')
+    state_deviating_time_unknown = Column(Boolean, server_default='false')
+    state_cancelled = Column(Boolean, server_default='false')
+
+    log_created_by_uuid = Column(UUID(as_uuid=True), ForeignKey("user.uuid"), nullable=True)
+    log_created_by_time = Column(DateTime, nullable=True)
+    log_modified_by_uuid = Column(UUID(as_uuid=True), ForeignKey("user.uuid"), nullable=True)
+    log_modified_by_time = Column(DateTime, nullable=True)
 
-    def __init__(self, title: str, description: str, time: datetime, event: Event):
+    def __init__(
+        self, 
+        event: Event, 
+        title: str, 
+        description: str,
+        location: str,
+        time: datetime, 
+        deviating_time: datetime, 
```

Comment:
If `deviating_time` is not something that should exist in a newly created agenda entry it should not exist in the constructor. probably. unless you expect there to be a need for it in the future.

Instead, initialize it to `None` in the constructor body

Reply:
`deviating_time` has been commented out and is no longer a part of the constructor since its not something the user can set when you create a new agenda entry. It can be set in the future if the agenda entry has got a deviating time.

As of now, the code works without `deviating_time` in the constructor set to `None`.
What is right to do? 

1. Not define any of the variables in the constructor, but define them in the json/data which I am doing right now?
[Check here](https://github.com/phoenixlan/phoenixapi-v1/blob/29875e824420843f01e100b8773d8b91c329f4de/phoenixRest/models/core/agenda_entry.py#L51-L101)
2. Should I define all variables that can be set in the future to `None` in the constructor, and keep current code in the json/data?
3. Or something else?

Reply:
3. All variables that need to be defined on creation should be defined in the constructor. For the rest, I recommend using SQL defaults!

Reply:
Did as you suggested!
[Check here](https://github.com/phoenixlan/phoenixapi-v1/commit/4aa7b48a811ee05b4b490385a57995c5528f8d4e?diff=split)